### PR TITLE
Lesson 17 -fercan

### PR DESCRIPTION
**Pull request by fers9021**

### CODE REVIEW

**Comment by fers9021** [comment]
finished

File: Lesson16/fercan/script.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// 3. Write a function that asks the user to enter two numbers and checks if the first number is greater than, less than, or equal to the second number using if else statement.
```
Comment (annasvst) [inline Lesson16/fercan/script.js:1]: In this exercise, you're asked to write a function, so you need to start with a function definition, e.g.:
```

function compareNumbers(number1, number2) {
// if statement should be here inside the function
}
```

In our case, 'asks the user to enter two numbers' means that the function should take 2 parameters (number1, number2) as added in the example above. 

After function definition is created, it's time to call the function with actual numbers:
```
compareNumbers(2, 10);
compareNumbers(-20, 0);
compareNumbers(23, 3);

```

File: Lesson16/fercan/script.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    console.log('It\'s a negative number');
+ } else if (number>0) {  
+    console.log('It\'s a positive number');
+ } else { 
+    console.log ('It\'s equal to zero.');
+}}
+// 6. Write a function that takes in a year and checks if it's a leap year (is divisible by 4 and not divisible by 100 OR divisible by 400) 
+//or not using if else statement.
+
+function isLeapYear (year) {
```
Comment (annasvst) [inline Lesson16/fercan/script.js]: The function does not work correctly now as if you use 'else' statement, you cannot provide a condition. https://www.w3schools.com/js/js_if_else.asp

You could do like this:
```
if(year%4==0 && year%100!=0){
        console.log('It\'s a leap year');
     } else if (year%400==0); {  
        console.log('It\'s a leap year');
    } else {
      console.log('It\'s not a leap year');
  }
}
```

or you can do it in a shorter way by combining 2 conditions with logical OR statement (||):
```
if ((year % 4 === 0 && year % 100 !== 0) || year % 400 === 0) {
        return `${year} is leap year`;
    } else {
        return `${year} is not a leap year`;
    }
```

File: Lesson16/fercan/script.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    } else {
+        console.log('You got an A+ - Great job!');
+    }
+}
+
+// 14. Write a function that asks the user to enter a grade letter (A, B, C, D, or F) and 
+//displays a message indicating whether the grade is passing (A, B, C) or failing (D, F). 
+//Use a switch statement to handle the different cases.
+
+let grade=0;
```
Comment (annasvst) [inline Lesson16/fercan/script.js:87]: Here function definition is missing, try to write the code similar to the first example

File: Lesson16/fercan/script.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        console.log('You passed the test, but your grade is not great');
+    } else if (71 <= number && number <= 80) {
+        console.log('You got a B');
+    } else if (81 <= number && number <= 90) {
+        console.log('You got an A');
+    } else {
+        console.log('You got an A+ - Great job!');
+    }
+}
+
```
Comment (annasvst) [inline Lesson16/fercan/script.js:82]: Add some function calls below the function definition, e.g. 

```
testFailOrPass(40);
testFailOrPass(70);
testFailOrPass(99);

```

**Review by annasvst — CHANGES_REQUESTED** [review]
Added comment fro lesson 16

File: Lesson17/fercan/script.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    }
+}
+        
+console.log('- - - - - - - - - - - - - - - - - ');
+
+
+// 3. Write a loop that prints out the even numbers from 1 to 20.
+// 2, 4, 6...
+
+for(let i=1; i<=10; i++){
```
Comment (annasvst) [inline Lesson17/fercan/script.js]: We need to print numbers till 20, so `i<=20`

File: Lesson17/fercan/script.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  }
+}
+console.log("The minimum number in the array is:", minNumber);
+console.log('- - - - - - - - - - - - - - - - - ');
+
+
+// 6. Write a loop that reverses a given string.
+const string = 'Lorem ipsum dolor sit amet consectetur adipisicing elit.';
+
+console.log('answer to number6: no answer - - - - - - - - - - - - ');
```
Comment (annasvst) [inline Lesson17/fercan/script.js:66]: I think we did this exercises in the lesson, feel free to look in the lesson notes

**Review by annasvst — CHANGES_REQUESTED** [review]

**Review by fers9021 — COMMENTED** [review]
fixed

**Review by annasvst — APPROVED** [review]
Approved

**Pull request merged**